crates/sui-types/src/multisig.rs

```diff
@@ -102,7 +102,9 @@ impl AuthenticatorTrait for MultiSig {
     where
         T: Serialize,
     {
-        if self.multisig_pk.pk_map.len() > MAX_SIGNER_IN_MULTISIG {
+        let max_signer_in_multisig =
+            ProtocolConfig::get_for_min_version().max_num_multisig_signers();
```

Comment:
you can't use `get_for_min_version()` here - you have to use the current version. Yes, that requires a lot of plumbing, but unfortunately its the only way.

Reply:
Right I wasn't too sure about this piece too.
For something like the keytool which can invoke this feature, the user will probably have to provide their own version info explicitly?

Reply:
Right, for keytool, it can default to the max version, and pass in the ProtocolConfig, and you can add a commandline flag if desired.

But this has to run on the validator as well, to enforce the limit, right? In that case you need to pass in the config that you get from the epoch store (example: https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority.rs#L951)

Reply:
is there a client api to get the protocol epoch store for this version? dont want to complicate your PR, we can just use a default for keytool. 

in validator this check is enforced here (invoking `.verify_secure`) which i think you should have epoch store access: https://github.com/MystenLabs/sui/blob/c5e27be10b84e917687b55200005e0c1bd8da915/crates/sui-core/src/safe_client.rs

on a separate note, this max signer limit would be strictly increasing with protocol version, so the limit itself will always be backward compatible

Reply:
After mainnet it will indeed be probably strictly increasing (and this is what you care about). In theory there are ways to reduce it for new accounts, but unlikely we'll do it unless there is a huge reasons to do so. Before mainnet we still have time to check if N = 10 will be the final choice based on DoS protection analysis.